src/pepr/operator/controllers/network/authorizationPolicies.ts

```diff
@@ -0,0 +1,269 @@
+
+// Constants for gateway principals.
+const ADMIN_INGRESS = "cluster.local/ns/istio-admin-gateway/sa/admin-ingressgateway";
+const TENANT_INGRESS = "cluster.local/ns/istio-tenant-gateway/sa/tenant-ingressgateway";
+
+/**
+ * Type guard to determine if a rule is an Expose rule.
+ */
+function isExposeRule(rule: Allow | Expose): rule is Expose {
+  return "service" in rule && Boolean(rule.service);
+}
+
+/**
+ * Derives a short name from a selector.
+ */
+function derivePolicyName(selector: { [key: string]: string }): string {
+  if (selector["app"]) return selector["app"].replace(/-pod$/, "");
+  if (selector["app.kubernetes.io/name"])
+    return selector["app.kubernetes.io/name"].replace(/-pod$/, "") + "-workload";
+  return "workload";
+}
+
+/**
+ * Generates a unique name for an Allow rule.
+ * Uses the description if provided; otherwise a combination of the selector values
+ * and remote properties is used.
+ */
+function generateAllowName(rule: Allow): string {
+  const { description, selector, remoteGenerated, remoteNamespace, remoteSelector } = rule;
+  const baseName =
+    description ||
+    [
+      Object.values(selector || { default: "all pods" }).join("-"),
+      remoteGenerated || [
+        remoteNamespace || "default",
+        Object.values(remoteSelector || { default: "all pods" }).join("-"),
+      ],
+    ]
+      .flat()
+      .join("-");
+  return `ingress-${baseName}`.toLowerCase().replace(/\s+/g, "-");
+}
+
+/**
+ * Generates a unique name for an Expose rule using effective port, selector, and gateway.
+ */
+function generateExposeName(rule: Expose): string {
+  const effectivePort = rule.targetPort ?? rule.port;
+  const selPart = rule.selector ? Object.values(rule.selector).join("-") : "all";
+  const gateway = rule.gateway || "tenant";
+  return `ingress-${effectivePort}-${selPart}-istio-${gateway}-gateway`
+    .toLowerCase()
+    .replace(/\s+/g, "-");
+}
+
+/**
+ * Generates a unique authorization policy name for a given rule.
+ */
+export function generateAuthPolName(rule: Allow | Expose): string {
+  return isExposeRule(rule) ? generateExposeName(rule) : generateAllowName(rule);
+}
+
+/**
+ * Processes an Allow rule to extract its effective source and ports.
+ */
+function processAllowRule(rule: Allow, pkgNamespace: string): { source: Source; ports: string[] } {
+  const ports: string[] = [];
+  if (rule.port !== undefined) ports.push(rule.port.toString());
+  if (rule.ports) ports.push(...rule.ports.map(p => p.toString()));
+
+  let source: Source = {};
+  if (rule.remoteCidr) {
+    source = { ipBlocks: [rule.remoteCidr] };
+  } else if (
+    rule.remoteNamespace === "" ||
+    rule.remoteNamespace === "*" ||
+    rule.remoteGenerated === RemoteGenerated.Anywhere
+  ) {
+    source = {};
+  } else if (rule.remoteGenerated === RemoteGenerated.IntraNamespace) {
+    source = { namespaces: [pkgNamespace] };
+  } else if (rule.remoteNamespace && rule.remoteNamespace !== "" && rule.remoteNamespace !== "*") {
```

Comment:
nit: I think this is redundant since it's an else if after the second two conditions have been checked (in the inverse) already?
```suggestion
  } else if (rule.remoteNamespace) {
```

Reply:
yeah good eyes